src/main.rs

```diff
@@ -78,26 +84,16 @@ fn main() {
         }
     }));
 
-    let arg = ::std::env::args().nth(1);
-    match arg.as_ref().map(|s| &s[..]) {
-        Some("--winit") => {
-            info!("Starting magmawn with winit backend");
-            winit::init_winit();
-        }
-        Some("--tty-udev") => {
-            info!("Starting magma on a tty using udev");
-            udev::init_udev();
-        }
-        Some(other) => {
-            error!("Unknown backend: {}", other);
-            info!("Possible backends are:");
-            for line in POSSIBLE_BACKENDS {
-                println!("{}", line);
-            }
-        }
-        None => {
-            backends::init_backend_auto();
-        }
+    let args = Args::parse();
+
+    if args.winit {
```

Comment:
can this be replace with a match case

Reply:
Not as far as I know. It's not an enum, it's two separate bools. The user could hypothetically pass both, so matching doesn't make sense here. (I have chosen to ignore that edge case - if they pass both it will use winit).

Reply:
i think tuples might work for that

Reply:
Is there no way for clap to have mutually exclusive arguments

Reply:
something like this should work

```rust
match (a, b) {
    (true, false) => println!("tf"),
    (true, true) => println!("tt"),
    (false, true) => println!("ft"),
    (false, false) => println!("ff"),
}
```

Reply:
There may be a way, but I am not aware of it. If we are concerned about it we can add a check in the code an print a special error message if both are passed. I thought that since the previous implementation also did not worry about this case that it wasn't a big concern.

Reply:
its not a big deal imo, if hackos is fine with this imo this can be resolved

Reply:
Another way would be something like `--backend=winit` or `--backend=tty-udev`, that can definitely be mutually exclusive.

Reply:
> something like this should work
> 
> ```rust
> match (a, b) {
>     (true, false) => println!("tf"),
>     (true, true) => println!("tt"),
>     (false, true) => println!("ft"),
>     (false, false) => println!("ff"),
> }
> ```

I could do that. Let me know if you would prefer this approach or changing it to a `--backend=` flag.

Reply:
either works, up to you

Reply:
I personally prefer the `--backend=` approach. @HackedOS are you on board with that?

Reply:
```#[derive(Parser)]
#[clap(author, version, about, long_about = None)]
struct Cli {
    /// The PO csv file to be used
    #[clap(short, long)]
    input: PathBuf,
    /// The destination directory where the processed POs will be saved
    #[clap(short, long)]
    output: PathBuf,
    /// The text file that contains all of the store numbers to be processed
    #[clap(short, long)]
    list: PathBuf,
    /// Print all RFIDs including items marked with a '$'
    #[clap(short = 'a', long = "print-all", conflicts_with = "report")]
    printall: bool,
    /// Produce a report of selected PO
    #[clap(short, long, conflicts_with = "printall")]
    report: bool,
}``` deriving `conflicts_with` according to clap docs

Reply:
Oh cool. Well, I've changed everything now so it won't be necessary for this one.